packages/flutter/lib/src/animation/curves.dart

```diff
@@ -267,6 +273,526 @@ class Cubic extends Curve {
+    // subdivisions, whichever comes first.
+    const double errorLimit = 1e-6;
+    int count = 100;
+    final double startValue = offsetToOrigin(start);
+    while ((end - start) / 2.0 > errorLimit && count > 0) {
+      mid = (end + start) / 2.0;
+      final double value = offsetToOrigin(mid);
+      if (value.sign == startValue.sign) {
+        start = mid;
+      } else {
+        end = mid;
+      }
+      count--;
+    }
+    return mid;
+  }
+}
+
+/// A curve that passes smoothly through the given control points using a
+/// centripetal Catmull-Rom spline.
+///
+/// When this curve is evaluated with [transform], the values will interpolate
+/// smoothly from one control point to the next, passing through (0.0, 0.0), the
+/// given points, and then (1.0, 1.0).
+///
+/// {@macro flutter.animation.curves.catmull_rom_description}
+///
+/// This class uses a centripetal Catmull-Rom curve (a [CatmullRomSpline]) as
+/// its internal representation. The term centripetal implies that it won't form
+/// loops or self-intersections within a single segment, and corresponds to a
+/// Catmull-Rom α (alpha) value of 0.5.
+///
+/// See also:
+///
+///  * [CatmullRomSpline], the 2D spline that this curve uses to generate its values.
+///  * A Wikipedia article on [centripetal Catmull-Rom splines](https://en.wikipedia.org/wiki/Centripetal_Catmull%E2%80%93Rom_spline).
+///  * [new CatmullRomCurve] for a description of the constraints put on the
+///    input control points.
+///  * This [paper on using Catmull-Rom splines](http://faculty.cs.tamu.edu/schaefer/research/cr_cad.pdf).
+class CatmullRomCurve extends Curve {
+  /// Constructs a centripetal [CatmullRomCurve].
+  ///
+  /// It takes a list of two or more points that describe the points that the
+  /// curve must pass through. See [controlPoints] for a description of the
+  /// restrictions placed on control points.
+  ///
+  /// The optional [tension] argument controls how tightly the curve approaches
+  /// the given `points`. It must be in the range 0.0 to 1.0, inclusive. It
+  /// defaults to 0.0, which provides the smoothest curve. A value of 1.0
+  /// is equivalent to a linear interpolation between points.
+  ///
+  /// All of the arguments must not be null.
+  ///
+  /// See also:
+  ///
+  ///  * This [paper on using Catmull-Rom splines](http://faculty.cs.tamu.edu/schaefer/research/cr_cad.pdf).
+  CatmullRomCurve(this.controlPoints, {this.tension = 0.0})
+      : assert(tension != null),
+        assert(() {
+          return validateControlPoints(
+            controlPoints,
+            tension: tension,
+            reasons: _assertReasons..clear(),
+          );
+        }(), 'control points $controlPoints could not be validated:\n  ${_assertReasons.join('\n  ')}'),
+        // Pre-compute samples so that we don't have to evaluate the spline's inverse
+        // all the time in transformInternal.
+        _precomputedSamples = CatmullRomSpline(
```

Comment:
Could/Should we do this lazily the first time we need it?

Reply:
I made a `.precompute` constructor here too.